ENG-4983 Amend the ComponentManager endpoint for delete POST /components/{bundleCode}/uninstall

entgigi opened the pull request.

entgigi committed:
ENG-4983 added amend for the delete rest service and the call to the new appEngine delete all components service

mirkoPan commented on src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java:
```diff
@@ -208,4 +228,34 @@ private JobResult executeUninstall(Installable<?> installable) {
+        }
+    }
+
+    private EntandoCoreComponentDeleteRequest convertEntityToDto(EntandoBundleComponentJobEntity entity) {
```
Consider putting this method in the dto in order to have greater uniformity in the mapping process than solutions adopted elsewhere in the code. On the other hand, the coupling of this suggestion, which would bind the entity with the dto, must be considered,

entgigi replied:
done, pls resolve if it's ok for you

mirkoPan commented on src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java:
```diff
@@ -208,4 +228,34 @@ private JobResult executeUninstall(Installable<?> installable) {
+        if (!EntandoCoreComponentDeleteResponseStatus.SUCCESS.equals(response.getStatus())) {
+            // FIXME what to do with failure status ???
+            try {
+                String uninstallErrors = om.writeValueAsString(response.getComponents().stream()
```
consider using a more descriptive variable

entgigi replied:
done, pls resolve if it's ok for you

mirkoPan reviewed (approved):
LGTM

firegloves commented on src/main/java/org/entando/kubernetes/client/core/DefaultEntandoCoreClient.java:
```diff
@@ -632,6 +636,47 @@ public EntandoCoreComponentUsage getComponentUsage(String code, String[] endpoin
+                                    new ParameterizedTypeReference<SimpleRestResponse<EntandoCoreComponentDeleteResponse>>() {
+                                    });
+
+            var usage = (ResponseEntity<SimpleRestResponse<EntandoCoreComponentDeleteResponse>>) retryer.execute(r);
```
Are we deleting data or fetching usage details? Why is the var called `usage`?

entgigi replied:
done

firegloves commented on src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java:
```diff
@@ -127,9 +136,10 @@ private void submitUninstallAsync(EntandoBundleJobEntity parentJob, EntandoBundl
                 scheduler.queuePrimaryComponents(uninstallJobs);
 
-                JobProgress uninstallProgress = new JobProgress(1.0 / uninstallJobs.size());
+                JobProgress uninstallProgress = new JobProgress(1.0 / (uninstallJobs.size() + 1));
```
Please add a comment to explain why we need the new `+ 1`

entgigi replied:
done

firegloves commented on src/test/java/org/entando/kubernetes/utils/TestInstallUtils.java:
```diff
@@ -1132,21 +1143,17 @@ public static String simulateInProgressUninstall(MockMvc mockMvc, EntandoCoreCli
         Mockito.reset(coreClient);
         WireMock.reset();
-        WireMock.setGlobalRandomDelay(delayDistribution);
+        //WireMock.setGlobalRandomDelay(delayDistribution);
```
please delete this commented line

entgigi replied:
done

firegloves commented on src/test/java/org/entando/kubernetes/controller/mockmvc/InstallFlowTest.java:
```diff
@@ -325,56 +324,42 @@ public void shouldCallCoreToUninstallComponents() throws Exception {
-        verify(coreClient, times(2)).setPageStatus(ac.capture(), ac.capture());
-        assertThat(ac.getAllValues()).containsAll(Arrays.asList("my-page", "another-page"));
-        assertThat(ac.getAllValues()).contains("draft");
-
-        ac = ArgumentCaptor.forClass(String.class);
-        verify(coreClient, times(2)).deletePage(ac.capture());
-        assertThat(ac.getAllValues()).containsAll(Arrays.asList("my-page", "another-page"));
-
+        */
```
please delete these commented lines

entgigi replied:
done

entgigi committed:
ENG-4983 added changes from code review

firegloves reviewed (approved):
LGTM

entgigi committed:
ENG-4983 fixed wrong message

mirkoPan committed:
ENG-4983 changed app-engine delete endpoint

entgigi committed:
ENG-4983 added fix to remove check for in use

entgigi committed:
ENG-4983 added fix to translate templateModel to pageModel, added changes from new specs and changed and fixed failure recording with a better mark for error

entgigi committed:
ENG-4983 set bundle as removed only for app-engine uninstall success,
         added more logs and fix for WidgetDescriptor
         filtered components by usage
         added fix to use only existing components in core client request for delete enabling subsequents retries
         added more test
         fixed integration test wrong expected value
         added optimization on error management

entgigi committed:
ENG-5050 fixed translation code and added a stronger object representation with enum

entgigi committed:
ENG-4983 fixed codestyle and tests
         added changes from sonar complaints
         removed duplicate code and made the uninstallation process behave the same for both uninstall and rollback (special case of uninstall)

mirkoPan committed:
ENG-4983 code smell fix

entgigi committed:
ENG-4983 added fix for deletion order
